src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -1510,4 +1590,48 @@ private static String prepareNumberFromString(String numberText) {
 
         return numberText;
     }
+    
+    /**
+     * @return true if input String can be parsed as 32-bit hex
+     */
+    private static boolean is32BitHex(String value) {
+    	return value.matches("0x[0-9a-fA-F]{8}");    	
+    }
+    
+    /**
+     * @return true if input String can be parsed as 64-bit hex
+     */
+    private static boolean is64BitHex(String value) {
+    	return value.matches("0x[0-9a-fA-F]{16}");    	
+    }
+     
+    private static float parseFloatOrFixed32(String value) {    	

```

Comment:
Can you explain why this and the `parseDoubleOrFixed64` methods are needed?  Add Javadoc, which explains it.